pkg/controllers/machine/disruption/drift.go

```diff
@@ -100,3 +102,13 @@ func areStaticFieldsDrifted(provisioner *v1alpha5.Provisioner, machine *v1alpha5
 	}
 	return provisionerHash != machineHash
 }
+
+func areNodeRequirementsDrifted(provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) bool {
+	return !lo.EveryBy(provisioner.Spec.Requirements, func(provisionerReq v1.NodeSelectorRequirement) bool {
+		return lo.ContainsBy(machine.Spec.Requirements, func(machineReq v1.NodeSelectorRequirement) bool {
+			return provisionerReq.Key == machineReq.Key &&
+				provisionerReq.Operator == machineReq.Operator &&
+				lo.Every(provisionerReq.Values, machineReq.Values)
```

Comment:
I don't think this is right. Wouldn't this check if there's a node requirement that matches each provisioner requirement? For instance, If the provisioner requirement is `Key Exists` and the Machine requirement is `Key In Values =[a,b,c]`, this logic would say that's not compatible, when it is.

This also assumes incorrectly that each requirement will have a set of values.

Reply:
I think the work is more complicated than this, and can use the requirements logic too. 

As a thought exercise, you should be able to check the cartesian product of all the possible Operators for a Provisioner and for a Machine

https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core#NodeSelectorOperator